modelcheck/src/main/kotlin/de/itemis/mps/gradle/modelcheck/ModelCheck.kt

```diff
@@ -136,64 +135,69 @@ fun writeJunitXml(modules: Iterable<SModule>,
 
     val modelsToReport = models.union(errorsPerModel.keys)
 
-    val xmlMapper = XmlMapper()
-
-    @Suppress("DEPRECATION")
-    when (format) {
-        ReportFormat.ONE_TEST_PER_MODEL -> {
-            val message = "The option `--result-format model` is deprecated as it doesn't report module level errors, " +
-                    "please use `--result-format module-and-model` instead."
-            printError(message)
-            System.err.println(message)
-
-            val testsuite = Testsuite(name = "model checking",
+    val value: Any =
```

Comment:
Why work with `Any` here? We do know there are only two types: `Testsuite` or `Testsuites` and thus we can correctly type here. 

My idea would be: The entire `value` assignment can be dropped and the `toDocument` call can instead be moved into each `when` condition, thus calling the correctly typed function.

A common meta class could also be an alternative.

Reply:
I don't care what type this thing has as long as I can pass it to toDocument. Instead of a metaclass, a change to consider would be to just work with a list of Testsuite objects (drop Testsuite**s**) and have the serializer add the wrapping `<testsuites>` element only when there's more than one test suite in the list.